Remora.Discord.Commands/Conditions/RequireContextCondition.cs

```diff
@@ -68,21 +69,42 @@ public async ValueTask<Result> CheckAsync(RequireContextAttribute attribute, Can
 
             var channel = getChannel.Entity;
 
-            return attribute.Context switch
+            if (attribute.ChannelTypes != null)
             {
-                ChannelContext.DM => channel.Type is DM
+                return attribute.ChannelTypes.Contains(channel.Type)
                     ? Result.FromSuccess()
-                    : new ConditionNotSatisfiedError("This command can only be used in a DM."),
-                ChannelContext.GroupDM => channel.Type is GroupDM
-                    ? Result.FromSuccess()
-                    : new ConditionNotSatisfiedError("This command can only be used in a group DM."),
-                ChannelContext.Guild =>
-                    channel.Type is GuildText or GuildVoice or GuildCategory or GuildNews or GuildStore or
-                        GuildPrivateThread or GuildPublicThread
+                    : new ConditionNotSatisfiedError($"This command was invoked in a channel of type '{channel.Type}', but it can only be used in '{string.Join(", ", attribute.ChannelTypes.Select(x => x.ToString()))}'");
+            }
+
+            if (attribute.ChannelContexts == null)
+            {
+                throw new ArgumentException($"Neither {nameof(attribute.ChannelContexts)} nor {nameof(attribute.ChannelTypes)} are defined");
+            }
+
+            foreach (var channelContext in attribute.ChannelContexts)
+            {
+                var conditionResult = channelContext switch
+                {
+                    ChannelContext.DM => channel.Type is DM
+                        ? Result.FromSuccess()
+                        : new ConditionNotSatisfiedError("This command can only be used in a DM."),
+                    ChannelContext.GroupDM => channel.Type is GroupDM
+                        ? Result.FromSuccess()
+                        : new ConditionNotSatisfiedError("This command can only be used in a group DM."),
+                    ChannelContext.Guild => channel.Type is GuildText or GuildVoice or GuildCategory or GuildNews
```

Comment:
There's probably an argument to be made for moving these groupings out into an extension method on `ChannelContext` or a dictionary of some kind.